### PR TITLE
WW-2 CNF clickable dialog

### PR DESCRIPTION
**Pull request by nikodamn**
Add tests to verify if whole NCF dialog is clickable and redirects to Community Page.

Reviewers:
@Wikia/west-wing

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java [hunk 1]
```diff
@@ -11,27 +13,57 @@
...
-    entryPointButton.click();
+    return new SpecialCommunity();
+  }
 
-    return true;
+  public SpecialCommunity clickDialogImage() {
+    try {
+      wait.forElementVisible(image);
+      image.click();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java [hunk 2]
```diff
@@ -11,27 +13,57 @@
...
     return true;
   }
 
-  public boolean clickEntryPointButton() {
-    isVisible();
+  public SpecialCommunity clickHelpOutButton() {
+    try {
+      wait.forElementVisible(helpOutButton);
+      helpOutButton.click();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java [hunk 3]
```diff
@@ -11,27 +13,57 @@
...
+    }
+
+    return new SpecialCommunity();
+  }
+
+  public SpecialCommunity clickDialogContent() {
+    try {
+      wait.forElementVisible(dialogContent);
+      dialogContent.click();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java [hunk 4]
```diff
@@ -1,37 +1,70 @@
...
-    wait.forElementVisible(salesPitchDialog);
+  @FindBy(css = ".community-page-benefits-content")
+  private WebElement dialogContent;
 
-    return true;
+  public boolean isDialogVisible() {
+    try {
+      wait.forElementVisible(salesPitchDialog, 5);
+      return true;
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java [hunk 5]
```diff
@@ -1,37 +1,70 @@
...
+      PageObjectLogging.logInfo("Dialog is not visible");
+      return false;
+    } catch (TimeoutException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java [hunk 6]
```diff
@@ -1,37 +1,70 @@
...
+    return new SpecialCommunity();
   }
 
-  public boolean clickEntryPointButton() {
-    isVisible();
+  public SpecialCommunity clickDialogImage() {
+    try {
+      wait.forElementVisible(image);
+      image.click();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java [hunk 7]
```diff
@@ -1,37 +1,70 @@
...
+    }
   }
 
-  public boolean isNotVisible() {
-    wait.forElementNotVisible(salesPitchDialog);
+  public SpecialCommunity clickHelpOutButton() {
+    try {
+      wait.forElementVisible(helpOutButton);
+      helpOutButton.click();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java (hunk 4 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java [hunk 8]
```diff
@@ -1,37 +1,70 @@
...
+    return new SpecialCommunity();
+  }
 
-    wait.forElementVisible(entryPointButton);
-    entryPointButton.click();
+  public SpecialCommunity clickDialogContent() {
+    try {
+      wait.forElementVisible(dialogContent);
+      dialogContent.click();
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java (hunk 5 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.

**Comment by idradm** [comment]
LGTM

**Pull request merged**